apps/okreads-e2e/src/specs/reading-list.spec.ts

```diff
@@ -17,4 +17,107 @@ describe('When: I use the reading list feature', () => {
       )
     );
   });
+
+  it('Then: I shall be able to undo adding a book to the reading list', async () => {
+    await browser.get('/');
+    await browser.wait(
+      ExpectedConditions.textToBePresentInElement($('tmo-root'), 'okreads')
+    );
+
+    // search for 'haskell' books
+    const form = await $('form');
+    const input = await $('input[type="search"]');
+    await input.sendKeys('haskell');
+    await form.submit();
+
+    // current reading list
+    const readingListToggle = await $('[data-testing="toggle-reading-list"]');
+    await readingListToggle.click();
+    await browser.wait(
+      ExpectedConditions.textToBePresentInElement(
+        $('[data-testing="reading-list-container"]'),
+        'My Reading List'
+      )
+    );
+    const itemsBeforeUndo = await $$('.reading-list-item');
+    const closeReadingList = await $('h2 button.mat-icon-button');
+    await closeReadingList.click();
+
+    // add new book
+    const addButton = await $('button[test-id="add-to-reading-list"][ng-reflect-disabled="false"]');
+    await addButton.click();
+    
+    // to be able to locate snackbar element
+    // otherwise, protractor stops to wait for snackbar to time out and disappear
+    await browser.waitForAngularEnabled(false);
+
+    const undoAddButton = await $('#snack-bar-button');
```

Comment:
Some inconsistency here. You're awaiting this, but not the toggle on line 58? I don't think you actually need to await this if you're only using the click action.